internal/core/testing.go

```diff
@@ -41,7 +41,7 @@ type TestResult struct {
 }
 
 // TestCheck is a function that perform assertion on a TestResult
-type TestCheck func(*testing.T, *TestResult)
+type TestCheck func(t *testing.T, result *TestResult, meta map[string]interface{})
```

Comment:
We should rename `TestResult` to `CheckContext` following the patter of `BeforeFuncCtx` and `AfterFuncCtx`